src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -101,34 +101,57 @@ public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $opt
 
 			$scalarTypes = TypeUtils::getConstantScalars($offsetType);
 			if (count($scalarTypes) === 0) {
-				$integerRanges = TypeUtils::getIntegerRanges($offsetType);
+				$integerRangesBackup = TypeUtils::getIntegerRanges($offsetType);
+				$integerRanges = $integerRangesBackup;
 				if (count($integerRanges) > 0) {
 					foreach ($integerRanges as $integerRange) {
-						if ($integerRange->getMin() === null) {
+						$minRange = $integerRange->getMin();
+						if ($minRange === null) {
 							break;
 						}
-						if ($integerRange->getMax() === null) {
+
+						$maxRange = $integerRange->getMax();
+                        if ($maxRange === null) {
 							break;
 						}
 
-						foreach (range($integerRange->getMin(), $integerRange->getMax()) as $rangeValue) {
+						$rangeValue = $minRange;
+						$rangeCount = 0;
+						do {
+							$rangeCount++;
+							if ($rangeCount > self::ARRAY_COUNT_LIMIT) {
+								$scalarTypes = $integerRangesBackup;
```

Comment:
I don't understand the logic. The `$scalarTypes` array should never contain IntegerRangeType, why do you assign it there?

Reply:
I am sure that the classes that triggers the issue #6375 contains int-ranges and that the `range` call triggers the fatal error. So I tried to test it via unit tests. 

What type triggers the fatal error from issue #6375? Does the int type somehow contains the int-range? I thought that this is a new type? 🤔